actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -92,10 +98,28 @@ def redirect_back(fallback_location:, allow_other_host: true, **args)
     #
     # All other options that can be passed to #redirect_to are accepted as
     # options and the behavior is identical.
-    def redirect_back_or_to(fallback_location, allow_other_host: true, **args)
-      referer = request.headers["Referer"]
-      redirect_to_referer = referer && (allow_other_host || _url_host_allowed?(referer))
-      redirect_to redirect_to_referer ? referer : fallback_location, **args
+    def redirect_back_or_to(fallback_location, allow_other_host: allow_open_redirects, **args)
+      referer = request.headers.fetch("Referer", fallback_location)
+      response_options = {
+        fallback_location: allow_open_redirects ? fallback_location : nil,
+        allow_other_host: allow_other_host,
+        **args,
+      }
+      redirect_to referer, response_options
```

Comment:
Don't think we should add `fallback_location` to `redirect_to`, if it's an unsafe redirect we should raise not just redirect to a fallback location. It's potentially an attacker trying to force a redirect so we should fail open and hit an error monitor.

Let's extract parts of `redirect_to` that we can reuse here and in `redirect_to`.

Reply:
It breaks some redirect tests if I remove the fallback location when permitting open redirects.

- `RedirectTest#test_safe_redirect_back_from_other_host`
- `RedirectTest#test_safe_redirect_back_with_no_referer_redirects_to_another_host`

I'm afraid we might break apps if we make this the default too early.

Reply:
Maybe we're talking past each other? I'm saying that `fallback_location` shouldn't be an argument that `redirect_to` accepts, that's what `redirect_back_or_to` is for.

Let's extract most of the `redirect_to` innards to a private `_redirect_to` method that we can then reuse in `redirect_back_or_to` as well.

Reply:
Ah, my mistake. I thought you were talking about the specific call site. I understand now, and agree 😄